packages/app/src/systems/Settings/hooks/useConnections.tsx

```diff
@@ -0,0 +1,176 @@
+import type { Account, Connection } from '@fuel-wallet/types';
+import { useInterpret, useSelector } from '@xstate/react';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+
+import type { ConnectionsMachineState } from '../machines';
+import {
+  ConnectionStatus,
+  ConnectionScreen,
+  connectionsMachine,
+} from '../machines';
+
+// ----------------------------------------------------------------------------
+// Selectors
+// ----------------------------------------------------------------------------
+
+function filteredOrAll<T>(
+  state: ConnectionsMachineState,
+  entity: 'accounts' | 'connections'
+): T {
+  const isAccounts = entity === 'accounts';
+  const { searchText = '' } = state.context.inputs;
+  const hasSearch = Boolean(searchText.length);
+  const response = state.context.response || {};
+  const all = response?.[isAccounts ? 'accounts' : 'connections'] || [];
+  const filteredProp = `filtered${isAccounts ? 'Accounts' : 'Connections'}`;
+  const filtered = response?.[filteredProp];
+  return hasSearch ? filtered : filtered || all;
+}
```

Comment:
I'm not a big fan of this type of abstraction, we lose tracking of props when accessing object dinamically, and also need to do a lot of conditional typing, conditional prop access

good
```
response.filteredAccounts
```

not good
```
  const filteredProp = `filtered${isAccounts ? 'Accounts' : 'Connections'}`;
  const filtered = response?.[filteredProp];
```

what you think on doing something like:
```
function filterAll<T = Account | Connection>(all: T[], filtered: T[], searchText: Maybe<string>): T[] {
  const hasSearch = Boolean(searchText.length);
  return hasSearch ? filtered : filtered || all;
}

function filterAccounts(all: Account[], filtered, searchText): Account[] {
  return filterAll<Account>(all, filtered, searchText);
}

function filterConnections(all, filtered, searchText): Connection[] {
  return filterAll<Connection>(all, filtered, searchText);
}

```

then in selectors:
```
const selectors = {
// ....
  accounts(state: ConnectionsMachineState) {
    return filteredAccounts(
      state.context.response?.accounts || [],
      state.context.response?.filteredAccounts || [],
      state.context.inputs.searchText
    );
  },

  connections(state: ConnectionsMachineState) {
    return filterConnections(
      state.context.response?.connections || [],
      state.context.response?.filteredConnections || [],
      state.context.inputs.searchText
    );
  },
  
```

Reply:
✅